lib/src/main/kotlin/konig/kontext/CoroutineKonigKontext.kt

```diff
@@ -0,0 +1,10 @@
+package org.konigsoftware.konig.kontext
+
+import com.google.protobuf.Message
+import kotlin.coroutines.AbstractCoroutineContextElement
+import kotlin.coroutines.CoroutineContext
+
+class CoroutineKonigKontext<ContextType : Message>(val context: ContextType) :
+    AbstractCoroutineContextElement(CoroutineKonigKontext) {
+    companion object Key : CoroutineContext.Key<CoroutineKonigKontext<*>>
```

Comment:
this kinda makes it look like you could have multiple different kontext's each of a different generic type, and they'd each get their own `Key` ?

I'm guessing that's not your intention (or even possible due to type erasure??) - in which case, maybe pull this out of the class?

could put it alongside the other two grpc key constants you have just to make clear how they all relate to each other (or don't - they look very similar but aren't actually related?) wdyt?

Reply:
This is a bit tricky. `CoroutineContext.Key` doesn't have a constructor so you can't move this outside of the companion object. We are also passing `CoroutineKonigKontext` into the parent class `AbstractCoroutineContextElement` constructor and so we also need the companion object defined

Reply:
I can try and see if there's a different way to do this another way, but this is how all the examples I've seen do it